### PR TITLE
RT-685 & 686 & 687 Setup Banxa API

### PR DESCRIPTION
**Pull request by iandjx**
# Description

Please include a summary of the changes and the related issue in checklist format. Please also include relevant motivation and context.

- [x] Added create-order API route to create new buy/sell transaction
- [x] Added crypto-buy API route to fetch all tokens that can be bought
- [x] Added crypto-sell API route to fetch all tokens that can be sold
- [x] Added fiat-buy and fiat-sell API route to fetch all fiat that can be bought or sold
- [x] Aded get-order and get-orders API route to fetch all or customer's orders
- [x] Added sample pages to test each route
- [x] Used firebase  to store email:account_reference mapping 
- [x] Use local storage to store the account_reference for future transactions

Please refer to https://docs.banxa.com/reference to check the args needed to be passed for API routes.

User flow is as follows:
1. User goes to RT
2. User opens banxa widget in an iframe
3. User selects either buy/sell order
4. User fills out required fields i.e. wallet address, source (fiat/token), target (fiat/token), email, amount-in
5. banxa widget will automatically update the amount to be received every time the amount-in changes
6. Upon submit, a new banxa order is created. the banxa order contains a checkout url
7. User will then be redirected to a new page/popup pointing to the  checkout url
8. User will complete the KYC, finalize payment details etc in banxa site
9. on completion, user will be redirected back to app.rometerminal.io

To test this user journey. 
- [x] go to create order
- [x] Click submit
- [x] copy the checkout URL 
- [x] open checkout URL in a new window
- [x] complete the payment process using dummy KYC details https://docs.banxa.com/docs/order-flow

## JIRA Issue
RT-685 , RT-686, RT-687

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
- [x] copy the banxa env variables i have sent to your email addresses
- [x]  rename .env.example to env
- [x] compile and run
- [x] check api responses as you switch pages

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: banxa/utils/banxa/generateHmac.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import crypto from "crypto";
+
+const generateHmac = (signature: string, nonce: number): string => {
+  const key = process.env.BANXA_API_KEY;
```
Comment (MarkCBall) [inline banxa/utils/banxa/generateHmac.ts:4]: where will this code live? in a server or in the client?

Comment (MarkCBall) [reply banxa/utils/banxa/generateHmac.ts:4]: i'd raise a red flag if its in the client as that means we expose our secrets

File: banxa/components/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const tabs = [
+  { name: "My Account", href: "#", current: false },
+  { name: "Company", href: "#", current: false },
+  { name: "Team Members", href: "#", current: true },
+  { name: "Billing", href: "#", current: false },
+];
+
+function classNames(...classes: string[]) {
```
[the remark this answers is not in the document]
Comment (iandjx) [reply banxa/components/Tabs.tsx]: deleted this

File: banxa/utils/banxa/generateHmac.ts (hunk 1 above)
Comment (iandjx) [reply banxa/utils/banxa/generateHmac.ts:4]: server. `process.env.` varaiables are different from `NEXT_PUBLIC_` , process.env variables can only be accessed by server side calls such as API routes

**Comment by iandjx** [comment]
Requires to switch to RBL's firebase account before switching to prod

**Pull request closed without merging**